File: kotlin/src/main/kotlin/com/authdog/sdk/exceptions.kt
package com.authdog.sdk

/**
 * Base exception for all Authdog SDK errors
 */
open class AuthdogError(message: String) : Exception(message)

/**
 * Raised when authentication fails
 */
class AuthenticationError(message: String) : AuthdogError(message)

/**
 * Raised when API requests fail
 */
class APIError(message: String) : AuthdogError(message)

File: node/src/exceptions.ts
/**
 * Base exception for all Authdog SDK errors
 */
export class AuthdogError extends Error {
  constructor(message: string) {
    super(message);
    this.name = 'AuthdogError';
  }
}

/**
 * Raised when authentication fails
 */
export class AuthenticationError extends AuthdogError {
  constructor(message: string) {
    super(message);
    this.name = 'AuthenticationError';
  }
}

/**
 * Raised when API requests fail
 */
export class APIError extends AuthdogError {
  statusCode?: number;

  constructor(message: string, statusCode?: number) {
    super(message);
    this.name = 'APIError';
    this.statusCode = statusCode;
  }
}
